x-pack/plugins/aiops/public/embeddables/change_point_chart/types.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import type { DefaultEmbeddableApi } from '@kbn/embeddable-plugin/public';
+import type {
+  HasEditCapabilities,
+  PublishesDataViews,
+  PublishesTimeRange,
+  PublishingSubject,
+  SerializedTimeRange,
+  SerializedTitles,
+} from '@kbn/presentation-publishing';
+import type { ChangePointDetectionViewType } from '@kbn/aiops-change-point-detection/constants';
+import type { SelectedChangePoint } from '../../components/change_point_detection/change_point_detection_context';
+
+export type ViewComponent = FC<{
+  changePoints: SelectedChangePoint[];
+  interval: string;
+  onRenderComplete?: () => void;
+}>;
+
+export interface ChangePointComponentApi {
+  viewType: PublishingSubject<ChangePointEmbeddableState['viewType']>;
+  dataViewId: PublishingSubject<ChangePointEmbeddableState['dataViewId']>;
+  fn: PublishingSubject<ChangePointEmbeddableState['fn']>;
+  metricField: PublishingSubject<ChangePointEmbeddableState['metricField']>;
+  splitField: PublishingSubject<ChangePointEmbeddableState['splitField']>;
+  partitions: PublishingSubject<ChangePointEmbeddableState['partitions']>;
+  maxSeriesToPlot: PublishingSubject<ChangePointEmbeddableState['maxSeriesToPlot']>;
+  updateUserInput: (update: ChangePointEmbeddableState) => void;
+}
+
+export type ChangePointEmbeddableApi = DefaultEmbeddableApi<ChangePointEmbeddableState> &
+  HasEditCapabilities &
+  PublishesDataViews &
+  PublishesTimeRange &
+  ChangePointComponentApi;
+
+export interface ChangePointEmbeddableState {
+  viewType: ChangePointDetectionViewType;
+  dataViewId: string;
+  fn: 'avg' | 'sum' | 'min' | 'max' | string;
+  metricField: string;
+  splitField?: string;
+  partitions?: string[];
+  maxSeriesToPlot?: number;
+}
+
+export type ChangePointEmbeddableRuntimeState = ChangePointEmbeddableState &
```

Comment:
Why are the time range and titles only part of the runtime state? Usually these would be part of the `SerializedState` because custom panel time ranges and panel titles are saved into the Dashboard.

Reply:
updated in https://github.com/elastic/kibana/pull/182246/commits/9d47fe5876d029c312c2cc019b72fb2ecc13f8cd